### ISSUE TITLE
Using inotify updates active layer only once when config is updated

### ISSUE THREAD

**Issue by AdityaGarg8** [WhatAmISupposedToPutHere/tiny-dfr#21]
Due to changes made in https://github.com/WhatAmISupposedToPutHere/tiny-dfr/commit/6689b2c3599e76be31d1b38b849f60d329a4e83d, if you update the config, the active layer gets updated only once. Upon subsequent updates to the config, you need to restart the service.

**Comment by WhatAmISupposedToPutHere** [comment]
Can you post detailed reproduction instructions? If i open the file with emacs, repeatedly edit and save it, it updates every time.

**Comment by AdityaGarg8** [comment]
> Can you post detailed reproduction instructions? If i open the file with emacs, repeatedly edit and save it, it updates every time.

I use gedit.

Open the file using gedit, edit MediaLayerDefault = false to true, save, the change occurs. Revert the change: no effect

Another instance, MediaLayerDefault = false to true, save, the change occurs. Remove bold from font, no change.

**Comment by AdityaGarg8** [comment]
I am able to reproduce this using emacs as well now.

**Comment by AdityaGarg8** [comment]
> Can you post the exact sequence of actions you are doing in emacs?

1. Run `sudo emacs /etc/tiny-dfr/config.toml`
2. Change `MediaLayerDefault = false` to `MediaLayerDefault = true`
3. Click save. The touchbar default layer changes.
4. Change `MediaLayerDefault = true` to `MediaLayerDefault = false`
5. No change occurs.
6. Change some random setting.
7. Again, no change.

**Comment by AdityaGarg8** [comment]
```
            if evt.wd == cfg_watch_desc {
                // Configuration file event
                println!("Inotify event: {:?}", evt);
```

I tried debugging, in the first change, it shows some messages, but in subsequent changes, no messages seen.

**Comment by AdityaGarg8** [comment]
Another issue, I was before starting the daemon after opening the config.toml file, if I open it when the daemon is running, it freezes.

With the debug added above, the terminal is filled with:

```
(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
Inotify event: InotifyEvent { wd: WatchDescriptor { wd: 1 }, mask: AddWatchFlags(IN_CLOSE_NOWRITE), cookie: 0, name: None }
```

**Comment by WhatAmISupposedToPutHere** [comment]
At least the freeze should be fixed now, try it.

**Comment by AdityaGarg8** [comment]
The latest commit fixed both my issues.

**Comment by AdityaGarg8** [comment]
I've noticed that although text editors work, using `sed` command in the file doesn't work yet. To make changes effective, I have to open config.toml using a text editor.

**Comment by AdityaGarg8** [comment]
I found out running `touch` afterwards fixes this.
